ccloud-monitoring/docs/consumer-scenarios/increasing-consumer-lag.rst

```diff
@@ -0,0 +1,168 @@
+producer's last produced message and the consumer group's last commit. If you are unfamiliar with
+consumer groups or concepts like committing offsets, please refer to this
+`Kafka Consumer documentation <https://docs.confluent.io/platform/current/clients/consumer.html>`__.
+
+A large consumer lag, or a quickly growing lag, indicates that the consumer is not able to keep up with
+the volume of messages on a topic.
+
+This scenario will look at |ccloud| metrics from the Metrics API, `kafka-lag-exporter <https://github.com/lightbend/kafka-lag-exporter>`__ metrics, and
+client metrics from the client application’s MBean object ``kafka.consumer:type=consumer-fetch-manager-metrics,client-id=<client_id>``.
+
+Introduce failure scenario
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. By default 2 consumers and 1 producer are running. Adjust that to 1 consumer and 5 producers in order to force the condition where the consumer cannot keep up with the rate of messages being produced, which will cause an increase in consumer lag.
+   The container scaling can be done with the command below:
+
+   .. code-block:: bash
+
+      docker-compose up -d --scale consumer=1 --scale producer=5
+
+   Which will produce the following output:
+
+   .. code-block:: bash
+
+      ccloud-exporter is up-to-date
+      kafka-lag-exporter is up-to-date
+      node-exporter is up-to-date
+      grafana is up-to-date
+      prometheus is up-to-date
+      Stopping and removing ccloud-monitoring_consumer_2 ... done
+      Starting ccloud-monitoring_producer_1              ... done
+      Creating ccloud-monitoring_producer_2              ... done
+      Creating ccloud-monitoring_producer_3              ... done
+      Creating ccloud-monitoring_producer_4              ... done
+      Creating ccloud-monitoring_producer_5              ... done
+      Starting ccloud-monitoring_consumer_1              ... done
+
+Diagnose the problem
+^^^^^^^^^^^^^^^^^^^^
+
+#. Open `Grafana <localhost:3000>`__ and login with the username ``admin`` and password ``password``.
+
+#. Navigate to the ``Consumer Client Metrics`` dashboard. Wait 2 minutes and then observe:
+
+   - ``Rebalance rate`` (``rebalance-rate-per-hour``) has a bump, indicating that the consumer group ``demo-cloud-monitoring-1`` underwent a rebalance, which is to be expected when a consumer leaves the group.
+
+   |Consumer Rebalance Bump|
+
+   - An upward trend in ``Consumer group lag in records``.  ``Consumer group lag in seconds`` will have a less dramatic increase.
+     Both indicate that the producer is creating more messages than the consumer can fetch in a timely manner.
+     These metrics are derived from the ``kafka-lag-exporter`` container.
+     `kafka-lag-exporter <https://github.com/lightbend/kafka-lag-exporter>`__ is a scala open source project that collects data about consumer groups and presents them in a Prometheus scrapable format.
```

Comment:
Great explanation!  For consideration, move to the intro?

Reply:
Are you sure? Earlier it seemed like you wanted an explanation of this sort here (https://github.com/confluentinc/examples/pull/948#discussion_r582325128).

Reply:
My leaning:

- In the introduction it could introduce the Kafka Lag Exporter overall (with the longer explanation to be able to answer what is Kafka Lag Exporter
- In this step it could say that consumer lag is derived from ``kafka-lag-exporter``.  User by this point should already know what Kafka Lag Exporter is, you are just reminding them that this metric. is derived from that

Reply:
I moved the kafka-lag-exporter info to the top in commit https://github.com/confluentinc/examples/pull/948/commits/9b6e040d6ad2773e51a2e0a49e50b944549ac902